README.md

```diff
@@ -73,7 +73,7 @@ Even though it is strongly preferred to rely on ECC P521 and SHA-512 where possi
 6. EnrollZ service obtains the oIAK and oIDevID certs from the CA and calls the device's `RotateOIakCert` API to persist the oIAK and oIDevID certs on the control card.
 7. The switch verifies that the IAK pub key in oIAK cert matches the one in IAK cert and that IDevID pub key in oIDevID cert matches the one in IDevID cert.
 8. The switch stores oIAK and oIDevID certs in non-volatile memory and will present them in the TPM attestation `attestz` workflow.
-9. The switch must update its default SSL profile to rely on the owner IDevID cert instead of IDevID cert.
+9. The switch must create a new SSL profile "oidevid_profile" to rely on the owner IDevID cert instead of IDevID cert.
```

Comment:
How is this new profile going to be configured? Is it going to be clear for the switch that it should rely on this new profile for initial attestz and certz? Do we need to supply any addition info to the switch other than Owner IDevID cert?

During bootz we also wanted to set Google's services' (e.g. enrollz service, PC, etc)  trustbundle on the device, so that the device could verify switch owners connection/identity (proper mutual TLS). Would that be reflected in SSL profile? If so, should we create this new profile during bootz instead?

Reply:
if you going to deliver the trust bundle as part of bootz but not the certs - then whatever SSL profile is created during bootz that refers to the trust bundle must be used to set the oidevid cert / key

Reply:
given that you want to use that same profile as part of enrollz - then i think you will need to add that profile id to he enrollz message

Reply:
bootz must create a new profile as it is providing a trust bundle.

there certainaly could be a scenario where bootz doesn't create a new policy but that is not a workflow we would support internally

enrollz "could create" such a profile but that would not be how enrollz is currently designed and would increase the complexity of enrollz would that be something you want to do?

there is no reason you need "two user defined" profiles you could if want but it wouldn't be required

you could during bootz make a profile called - "prod profile" rotate into that profile the cert bundle and use the idevid certs 

then during enrollz - rotate in the oidevid certs into that profile
then during certz - rotate in the prod certs while leaving the bundle / crl / auth policies all the same

or you could 

make a enrollz_profile during bootz - provide the trust bundle and use idevid certs
during enrollz rotate that profile's certs to oidevid 
then during certz - create a prod profile and either delete the enrollz profile or leave it... but not have any services use it...

i would leave it to users to define thier workflow - how we internally want to develop the workflow would be up to our service implementors

Reply:
"bootz must create a new profile as it is providing a trust bundle."
=> Does bootz create a new profile right now (the way it is designed and implemented atm)? If so, what configures the SSL profile ID and how is this new profile managed later?

"there certainly could be a scenario where bootz doesn't create a new policy but that is not a workflow we would support internally"
=> This would imply that enrollz does NOT create a *new* SSL profile, but instead updates the existing one specified by `ssl_profile_id` field. The existing profile must be set by either bootz (install scenario) or certz (post-install scenario). If so, please update the README accordingly.

"how we internally want to develop the workflow would be up to our service implementors"
=> Thanks, the above scenarios make sense to me and I agree that we should give users the flexibility with various ssl profiles. That being said, the change in this PR is not trivial to actually implement internally as we would need to coordinate bootz, enrollz and certz services to make sure they are using/creating/deleting right ssl profiles. Bootz service, for example, does not set Google's infra trust anchors on the device at the moment and from my understanding does not create a new SSL profile (happy to be corrected, so let me know!). Once it starts doing so, bootz (or higher level orchestrator) should start passing the newly created ssl profile ID to enrollz service (so that it updates the right profile) and certz service (so that it either (a) updates the existing profile or (b) creates a new profile and deletes an existing one). It is not a complicated work, but just something that needs (1) internal cross-team designing and coordination, (2) adjusting dev delivery timelines and (3) proper communication with all vendors about the new behavior. Once you address the feedback in the PR, please initiate those, and I would be happy to chime in, brainstorm and help with the impl.